### PR TITLE
Reject with error

### PR DESCRIPTION
**Pull request by AlexKamaev**
modified https://github.com/DevExpress/testcafe/pull/5784 PR

I moved the https://github.com/DevExpress/testcafe/pull/5784 to the separate PR, since I need to do rebase.

We need to throw error of Error type.
The test cathes the error correctly and logs it.

### CODE REVIEW

File: src/test-run/index.js [hunk 1]
```diff
@@ -954,6 +952,9 @@ ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
 
     this._clearPendingRequest();
 
+    if (this.disconnected)
```
Comment (miherlosev) [inline src/test-run/index.js:955]: I guess it better to move the error creation right after `this.debugLog.driverMessage(msg);` code line.

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]

**Review by AndreyBelym — APPROVED** [review]

File: src/errors/runtime/templates.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ export default {
     [RUNTIME_ERRORS.cannotRunLiveModeRunnerMultipleTimes]: 'Cannot run a live mode runner multiple times.',
     [RUNTIME_ERRORS.browserDisconnected]:                  'The {userAgent} browser disconnected. This problem may appear when a browser hangs or is closed, or due to network issues.',
     [RUNTIME_ERRORS.cannotRunAgainstDisconnectedBrowsers]: 'The following browsers disconnected: {userAgents}. Tests will not be run.',
+    [RUNTIME_ERRORS.testRunRequestInDisconnectedBrowser]:  'The requested "{browser}" browser was disconnected during the test execution',
```
Comment (DIRECTcut) [inline src/errors/runtime/templates.js:18]:
```suggestion
    [RUNTIME_ERRORS.testRunRequestInDisconnectedBrowser]:  'The requested "{browser}" browser has disconnected during the test execution',
```

Assuming that the browser disconnected at some point during the test.

**Review by DIRECTcut — CHANGES_REQUESTED** [review]

File: src/errors/runtime/templates.js (hunk 2 above)
Comment (AlexKamaev) [reply src/errors/runtime/templates.js]: Fixed, please review

File: src/errors/runtime/templates.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ export default {
     [RUNTIME_ERRORS.cannotRunLiveModeRunnerMultipleTimes]: 'Cannot run a live mode runner multiple times.',
     [RUNTIME_ERRORS.browserDisconnected]:                  'The {userAgent} browser disconnected. This problem may appear when a browser hangs or is closed, or due to network issues.',
     [RUNTIME_ERRORS.cannotRunAgainstDisconnectedBrowsers]: 'The following browsers disconnected: {userAgents}. Tests will not be run.',
+    [RUNTIME_ERRORS.testRunRequestInDisconnectedBrowser]:  'The requested "{browser}" browser has disconnected during the test execution',
```
Comment (DIRECTcut) [inline src/errors/runtime/templates.js:19]: I've asked Dirk. He suggests:
 
`"chrome:headless" has disconnected during test execution`

File: test/functional/fixtures/browser-provider/browser-reconnect/test.js [hunk 4]
```diff
@@ -71,6 +72,23 @@ describe('Browser reconnect', function () {
...
+                const proc = spawn(`node ${path.join(__dirname, 'run-log-error-on-disconnect-test.js')}`, { shell: true, env: { ...process.env, DEBUG: 'hammerhead:*' } });
+
+                proc.stderr.on('data', data => {
+                    errLog += data.toString('utf-8');
+                });
+
+                proc.on('close', resolve);
+            })
+                .then(() => {
+                    expect(errLog).contains('The requested "chrome:headless" browser has disconnected during the test execution');
```
Comment (DIRECTcut) [inline test/functional/fixtures/browser-provider/browser-reconnect/test.js:88]: see my previous comment

**Review by DIRECTcut — APPROVED** [review]

**Comment by AlexKamaev** [comment]
Closed in favor of https://github.com/DevExpress/testcafe/pull/5866

**Pull request closed without merging**